### ISSUE TITLE
Avoid .NET Standard 1.x dependencies in core stack repositories

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/arcade#13315]
Just recently, many of our core stack repositories received Component Governance issues because of references to vulnerable package versions. Most warnings were mitigated by upgrading the transitive dependencies but repositories can actually avoid such transitive dependencies all-together. Avoiding netstandard1.x dependencies results in a much smaller dependency graph which makes a) restore more efficient and b) reduces surface area for future vulnerabilities.

**Example**:
- The [`System.Net.Http` nuget package](https://www.nuget.org/packages/System.Net.Http) is referenced in many .NET Framework and .NETCoreApp builds in our stack.
- That package hasn't been updated in years and all versions except the current (4.3.4) are marked as vulnerable.
- With CG alerts, repositories decided to upgrade that dependency (which is referenced either directly or transitively) to the latest version. While that's a viable approach, it mitigates the current alert but doesn't help avoiding future alerts.

=> Better: As `System.Net.Http.dll` is inbox in both .NET Framework and .NETCoreApp, that package dependency can be avoided entirely.
- If the dependency is referenced directly on .NETCoreApp, simply remove it. It's already referenced implicitly via the `Microsoft.NetCore.App` targeting pack.
- If the dependency is referenced directly on .NETFramework, replace it with `<Reference Include="System.Net.Http" />` which brings in the assembly from the .NET Framework installation (GAC).
- If the dependency isn't directly referenced, check the app's project.assets.json file to find the dependency that brings it in transitively. Try to avoid that parent parent dependency by walking the graph.
---

[Arcade's FlagNetStandard1xDependencies switch](https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.Arcade.Sdk/tools/TargetFrameworkFilters.BeforeCommonTargets.targets#L21) validates that projects don't bring in the .NET Standard 1.x dependency graph which consists of de-facto deprecated and vulnerable packages. While some netstandard1.x dependencies can't be avoided, i.e. xunit doesn't provide a `netstandard2.0` TFM and won't until v3 ships, most existing (transitive) dependencies can actually be avoided.

Eventually, by removing all netstandard1.x dependencies in at least the source build graph, we can remove the netstandard1.x targeting packs and package TFMs from the [SBRP](https://github.com/dotnet/source-build-reference-packages) repository. That itself reduces source build's payload size (which our partners care about) and validates that the dependency graph is _modern_ (based on the assumption that netstandard1.x assemblies are usually many years old).

The tables below track enabling that switch in our core stack repositories.

### Part of the VMR
| Repository | Status | Note |
| --- | --- | --- |
| arcade | | |
| aspnetcore | | |
| cecil | | |
| command-line-api | | |
| deployment-tools | | |
| diagnostics | | |
| dotnet | | |
| emsdk | | |
| format | | |
| fsharp | | |
| installer | | |
| msbuild | | |
| nuget-client | | |
| package-source-build | | |
| razor | | |
| roslyn-analyzers | | |
| roslyn | | |
| runtime | | |
| sdk | | |
| source-build-externals | | |
| source-build-reference-packages | | |
| sourcelink | | |
| symreader | | |
| templating | | |
| test-templates | | |
| vstest | | |
| xdt | :white_check_mark: | https://github.com/dotnet/xdt/commit/fcff05d598308522dc7c328c61e456823cdbee78, https://github.com/dotnet/xdt/commit/8b75ac1423171779f583867ad0fed0457bab9516 |
| xliff-tasks | :white_check_mark: | https://github.com/dotnet/xliff-tasks/commit/2d18d93582529a08ec2ff6da2d2f5c8e9a6b5e26, https://github.com/dotnet/xliff-tasks/commit/20262c86d25ad9e4e4f985a4b44d24a5de15bd05 |

### Outside of the VMR
| Repository | Status | Note |
| --- | --- | --- |

cc @ericstj @mmitche @MichaelSimons @SteveMCarroll

**Comment by missymessa** [comment]
@ViktorHofer is this something you're currently working on?

**Comment by ViktorHofer** [comment]
Yes. I also presented that in Tactics a few weeks ago. I will make steady progress over the coming weeks.

**Comment by ViktorHofer** [comment]
Closing as completed. The source-build graph got rid of all netstandard1.x dependencies. Additionally, repositories using the .NET 10 SDK benefit from the NuGet Package Pruning feature which avoids downloading these assets altogether.
